client/app/components/hearings-list-for-milestones-list.js

```diff
@@ -44,6 +44,33 @@ export function checkVotesSubmitted(records = [], recommendationType) {
   return projectsWithVotes.length > 0;
 }
 
+// Check that at least ONE disposition has a truthy date & location field
+// this is used to conditionally display the entire sub-milestone, including the title.
+export function checkHearingsShown(records = []) {
+  const projectsWithHearings = records.filter(function(disp) {
+    if ((disp.dcpDateofpublichearing !== null) && (disp.showHearingDetails)) {
+      return disp.dcpPublichearinglocation && disp.dcpDateofpublichearing.toString();
+    } return null;
+  });
+
+  return projectsWithHearings.length > 0;
+}
+
+// Check that at least ONE disposition has truthy values for recommendation field
+// this is used to conditionally display the entire sub-milestone, including the title.
+export function checkVotesShown(records = [], recommendationType) {
+  const projectsWithVotes = records.filter(function(disp) {
+    if (disp.showRecommendationDetails) {
+      // null values and empty strings will NOT be returned
+      return disp.get(recommendationType);
+    }
+    return false;
+  });
+
+  return projectsWithVotes.length > 0;
+}
+
+
```

Comment:
In the new functions, remove the logic already held by  checkVotesSubmitted and checkHearingsSubmitted. You can then either include all four booleans in the template, or have the two new funcitons depend on checkVotesSubmitted and checkHearingsSubmitted. For example:
export function checkHearingsShown(records = []) {
  const projectsWithHearingToShow = records.filter(function(disp) {
    if (disp.showHearingDetails) {
        return true
    } return false;
  });

       const areHearingsSubmitted = this.checkHearingsSubmitted(records);
  return areHearingsSubmitted && projectsWithHearingToShow
}